packages/core/src/components/dropdown-button/dropdown-button.tsx

```diff
@@ -0,0 +1,145 @@
+import { Component, h, Prop, Watch, Element } from "@stencil/core";
+import clsx from "clsx";
+import { tabbable } from "tabbable";
+import { v4 as uuidv4 } from "uuid";
+
+function tabInPopup(host: Element, down: boolean) {
+  const tabs = tabbable(host);
+  const currentIndex = tabs.findIndex((e) => e === document.activeElement);
+  const nextIndex = down
+    ? currentIndex >= tabs.length - 1
+      ? 0
+      : currentIndex + 1
+    : currentIndex <= 0
+    ? tabs.length - 1
+    : currentIndex - 1;
+  (tabs[nextIndex] as HTMLElement)?.focus();
+}
+
+function escapePopup(host: Element) {
+  host.shadowRoot?.children[0]?.getElementsByTagName("button")[0]?.focus();
+  host.setAttribute("open", "false");
+}
+
+function closePopupOnLastTab(host: Element) {
+  if (tabbable(host).pop() === document.activeElement) {
+    host.setAttribute("open", "false");
+  }
+}
+
+@Component({
+  tag: "dso-dropdown-button",
+  shadow: true,
+})
+export class DropdownButton {
+  /**
+   * Whether the dropdown is open or closed.
+   * This attribute is reflected and mutable.
+   */
+  @Prop({ reflect: true, mutable: true })
+  open = false;
+
+  /**
+   * The text of the dropdown button.
+   * */
+  @Prop()
+  label: string = "";
+
+  /**
+   * The button style variant
+   */
+  @Prop()
+  variant: "primary" | "secondary" | "tertiary" = "primary";
+
+  @Element()
+  el: HTMLElement | undefined;
+
+  @Watch("open")
+  openWatch(open: boolean) {
+    if (open) {
+      this.openPopup();
+    } else {
+      this.closePopup();
+    }
+  }
+
+  id = uuidv4();
+
+  connectedCallback() {
+    if (this.open) {
+      this.openPopup();
+    }
+
+    if (!this.el) {
+      return;
+    }
+
+    for (const ul of Array.from(this.el.getElementsByTagName("ul"))) {
+      ul.setAttribute("aria-labelledby", this.id);
+    }
+  }
+
+  openPopup() {
+    this.el?.addEventListener("keydown", this.keyDownListener);
+  }
+
+  closePopup() {
+    this.el?.removeEventListener("keydown", this.keyDownListener);
+  }
+
+  keyDownListener(event: KeyboardEvent) {
+    if (event.defaultPrevented) {
+      return;
+    }
+
+    const currentTarget = event.currentTarget as Element;
```

Comment:
`currentTarget` zal altijd het element van de event listener zijn, kunnen de functies `tabInPopup`, `escapePopup` en `closePopupOnLastTab` niet beter instance methods zijn? Volgens mij is `currentTarget === this.el`

Reply:
En dan ben je ook af van `host.setAttribute("open", "false")` en kan je `this.open = false` doen

Reply:
En dan toch typecasten, ondanks dat je daar niet zo'n fan van bent?

Reply:
Liever niet.

Maar misschien begrijp ik je niet goed, kan je iets meer uitleg geven?

Reply:
Volgens mij heb ik inmiddels de juiste constructie gevonden hiervoor. Aangepst dus.